### PR TITLE
Fixed achievements not working after a disc change.

### PR DESCRIPTION
**Pull request by LillyJadeKatrin**

### CODE REVIEW

File: Source/Core/Core/AchievementManager.cpp [hunk 1]
```diff
@@ -863,7 +863,10 @@ void AchievementManager::ChangeMediaCallback(int result, const char* error_messa
 {
   AchievementManager::GetInstance().m_loading_volume.reset(nullptr);
   if (result == RC_OK)
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:865]: Is this the only "good" result?  Should `RC_HARDCORE_DISABLED` be an error?

Comment (iwubcode) [reply Source/Core/Core/AchievementManager.cpp:865]: I was answered on discord.  This can be resolved (still wish user could resolve issues..)

**Review by iwubcode — APPROVED** [review]

File: Source/Core/Core/AchievementManager.cpp (hunk 1 above)
Comment (LillyJadeKatrin) [reply Source/Core/Core/AchievementManager.cpp:865]: Actually, no, you're right - I should be setting the function there too.

**Comment by JosJuice** [comment]
Should we really be setting the memory function to MemoryVerifier while switching discs to begin with? What if the achievement logic tries to read from memory while the disc is being changed?

**Comment by LillyJadeKatrin** [comment]
Fair point. I was expecting this to give me a crash from a callback similar to the load case but evidently not. I'll keep my ears open in case anyone reports such a crash though,

**Review by JosJuice — APPROVED** [review]

**Pull request merged**